system/lib/pthread/library_pthread_stub.c

```diff
@@ -371,3 +382,12 @@ int sem_destroy(sem_t *sem) {
 }
 
 void __wait(volatile int *addr, volatile int *waiters, int val, int priv) {}
+
+// When pthreads is not enabled, we can't use the Atomics futex api to do proper
+// sleeps, so simulate a busy spin wait loop instead.
+void emscripten_thread_sleep(double msecs) {
+  double start = emscripten_get_now();
+  while (emscripten_get_now() - start < msecs) {
```

Comment:
this will be much less efficient than to sleep in JS, where we don't constantly call out from wasm to JS.

I suppose this doesn't matter much as this just does a busy-wait. but it may mean a very tiny lag in the new version. so it maybe is best to keep in JS.

Reply:
Yes, I agree.  I reverted this part and kept this function in JS